internal/osutils/autostart/autostart_linux.go

```diff
@@ -171,16 +102,25 @@ func isEnabled(exec string, opts Options) (bool, error) {
 }
 
 func autostartPath(name string, _ Options) (string, error) {
-	dir, err := prependHomeDir(autostartDir)
+	// do not report ~/.profile as installed, as it would be removed on uninstall.
+	return "", nil
+}
+
+// https://activestatef.atlassian.net/browse/DX-1677
+func legacyDisableOnDesktop(exec string, opts Options) error {
+	dir, err := prependHomeDir(".config/autostart")
 	if err != nil {
-		return "", errs.Wrap(err, "Could not find autostart directory")
+		return errs.Wrap(err, "Could not find autostart directory")
 	}
-	path := filepath.Join(dir, name)
+
+	path := filepath.Join(dir, exec)
```

Comment:
Are you sure this is the correct path? We were using `opts.LaunchFileName` before:

https://github.com/ActiveState/cli/pull/2456/files#diff-fe1d6fb736a8d64ee905da5161a57eae198ad600fe9e62e48390d0ad205dd88fL49

Reply:
The recent rework has the disable logic using "exec":
https://github.com/ActiveState/cli/blob/086ad2976bdd8d7b01b8e03305030d561cf883da/internal/osutils/autostart/autostart_linux.go#L117
... which is passed as "name":
https://github.com/ActiveState/cli/blob/086ad2976bdd8d7b01b8e03305030d561cf883da/internal/osutils/autostart/autostart_linux.go#L178
In other words, "LaunchFileName" is not used for disabling. @MDrakos This feels like it might be a bug; Is this OK?

Reply:
`LaunchFileName` is the `state-svc.desktop` file which we should be removing. The `exec` argument that is being passed to `autoStartPath` is the name of the state service executable. This looks correct to me.

Reply:
For the legacy desktop uninstall, we do want to focus on the LaunchFileName. I've updated the logic and performed some other cleanup so that the primary functions read more simply, and all errors are now wrapped.